tests/param/test_params.py

```diff
@@ -0,0 +1,408 @@
+    prior_settings=None,
+):
+    def mock_config(model: SupportedModels, loglik_kind="analytical"):
+        if model not in default_model_config:
+            return lambda param: (None, None)
+
+        defaults = default_model_config[model]["likelihoods"][loglik_kind]
+
+        priors = defaults["default_priors"]
+        bounds = defaults["bounds"]
+
+        return lambda param: (priors.get(param), bounds.get(param))
+
+    config = Mock(
+        spec=Config,
+        get_defaults=mock_config(model_name, loglik_kind),
+    )
+
+    model = Mock(
+        model_name=model_name,
+        spec=HSSM,
+        list_params=default_model_config[model_name]["list_params"] + ["p_outlier"],
+        has_lapse=True,
+        loglik_kind=loglik_kind,
+        global_formula=global_formula,
+        link_settings=link_settings,
+        prior_settings=prior_settings,
+        model_config=config,
+        response_c="c(rt, response)",
+        additional_namespace={},
+    )
+    return model
+
+
+include = [
+    {"name": "v", "prior": {"name": "Normal", "mu": 0, "sigma": 1}},
+    {"name": "a", "prior": bmb.Prior("HalfNormal", sigma=1)},
+]
+
+
+def kw():
+    return {
+        "z": UserParam(prior=bmb.Prior("Normal", mu=0, sigma=1)),
+        "other_param": 1,
+    }
+
+
+p_outlier = 0.5
+
+
+def test_collect_user_params():
+    kwargs = kw()
+    model = create_mock_model("ddm")
+    print(model.list_params)
+    params = collect_user_params(model, include, kwargs, p_outlier)
+
+    assert isinstance(params, dict)
+    assert "t" not in params
+    for param_name, param in params.items():
+        assert param_name in ["v", "a", "z", "t", "p_outlier"]
+        assert isinstance(param, UserParam)
+
+    assert params["v"].prior == {"name": "Normal", "mu": 0, "sigma": 1}
+    assert params["a"].prior == bmb.Prior("HalfNormal", sigma=1)
+    assert params["z"].prior == bmb.Prior("Normal", mu=0, sigma=1)
+    assert params["p_outlier"] == UserParam("p_outlier", prior=0.5)
+
+    # kwargs is modified in place
+    assert "other_param" in kwargs
+    assert "z" not in kwargs
+
+
+def test_make_param_from_user_param():
+    # 1. Test the case with no global formula or link settings
+    # We test the following three cases:
+    # 1.1. UserParam that is definitely a regression parameter
+    # 1.2. UserParam that is definitely a simple parameter
+    # 1.3. UserParam that is ambiguous
+    model = create_mock_model("ddm")
+
+    # 1.1. UserParam that is definitely a regression parameter
+    reg_user_param = UserParam(
+        name="a",
+        formula="a ~ x",
+    )
+
+    assert reg_user_param.is_regression
+
+    param = make_param_from_user_param(model, "a", reg_user_param)
+    assert isinstance(param, RegressionParam)
+    assert param.name == "a"
+    assert param.formula == "a ~ x"
+    assert param.prior is None
+    assert (
+        param.bounds
+        == default_model_config["ddm"]["likelihoods"]["analytical"]["bounds"]["a"]
```

Comment:
It's meant to be a large dictionary, so that one day it could be re-written as a JSON or YAML file. We also thought about using shared configs between the `hssm` package and `ssm-simulators` package. As I mentioned, we definitely welcome ideas on how to implement this differently, including `lazy` ways to delay loading all this information until specifically requested